src/core/ipv4/ip4.c

```diff
@@ -629,6 +779,17 @@ ip4_input(struct pbuf *p, struct netif *inp)
        list of configured netifs. */
     if (ip4_input_accept(inp)) {
       netif = inp;
+#if IP_NAPT
+#if IP_FORWARD
```

Comment:
Again I'd suggest compiling the `if()` block only if enabled
```suggestion
#if IP_FORWARD && IP_FORWARD_ALLOW_TX_ON_RX_NETIF
```